ome_zarr/io.py

```diff
@@ -29,52 +33,62 @@ class ZarrLocation:
 
     def __init__(
         self,
-        path: Union[Path, str, FSStore],
+        path: StoreLike,
         mode: str = "r",
         fmt: Format = CurrentFormat(),
     ) -> None:
         LOGGER.debug("ZarrLocation.__init__ path: %s, fmt: %s", path, fmt.version)
         self.__fmt = fmt
         self.__mode = mode
-        if isinstance(path, Path):
-            self.__path = str(path.resolve())
-        elif isinstance(path, str):
-            self.__path = path
-        elif isinstance(path, FSStore):
-            self.__path = path.path
-        else:
-            raise TypeError(f"not expecting: {type(path)}")
+
+        store_path = make_store_path(path, mode=mode)
+        self.__path = store_path.path
+
+        # if isinstance(path, Path):
+        #     self.__path = str(path.resolve())
+        # elif isinstance(path, str):
+        #     self.__path = path
+        # elif isinstance(path, Store):
+        #     self.__path = path.path
+        # else:
+        #     raise TypeError(f"not expecting: {type(path)}")
 
         loader = fmt
         if loader is None:
             loader = CurrentFormat()
-        self.__store: FSStore = (
-            path if isinstance(path, FSStore) else loader.init_store(self.__path, mode)
+        self.__store: Store = (
+            path if isinstance(path, Store) else loader.init_store(self.__path, mode)
         )
 
-        self.__init_metadata()
+    async def init_meta(self):
+        loader = self.__fmt
+        if loader is None:
+            loader = CurrentFormat()
+
+        await self.__init_metadata()
+        print("init self.__metadata", self.__metadata)
         detected = detect_format(self.__metadata, loader)
-        LOGGER.debug("ZarrLocation.__init__ %s detected: %s", path, detected)
-        if detected != fmt:
+        LOGGER.debug("ZarrLocation.__init__ %s detected: %s", self.__path, detected)
+        if detected != self.__fmt:
             LOGGER.warning(
-                "version mismatch: detected: %s, requested: %s", detected, fmt
+                "version mismatch: detected: %s, requested: %s", detected, self.__fmt
             )
             self.__fmt = detected
-            self.__store = detected.init_store(self.__path, mode)
+            self.__store = detected.init_store(self.__path, self.__mode)
             self.__init_metadata()
 
-    def __init_metadata(self) -> None:
+    async def __init_metadata(self) -> None:
         """
         Load the Zarr metadata files for the given location.
         """
-        self.zarray: JSONDict = self.get_json(".zarray")
-        self.zgroup: JSONDict = self.get_json(".zgroup")
+        self.zarray: JSONDict = await self.get_json(".zarray")
```

Comment:
Instead of loading the json files directly, can you work with the metadata objects from zarr-python?